### PR TITLE
docs/HACKING.md: how to build systemd

### PR DESCRIPTION
**Pull request by beviu**
Explain what commands to run to build and test systemd for people that
are not familiar with meson.

Personally when I want to contribute to a project, a search for a file that explains how to build it. I did not find this information for systemd so I think it would be a good idea to add a paragraph in the HACKING.md file to explain how to build it.

### CODE REVIEW

File: docs/HACKING.md [hunk 1]
```diff
@@ -10,6 +10,17 @@ We welcome all contributions to systemd. If you notice a bug or a missing
...
 
+systemd uses the meson build system. meson works by generating build files to a
+separate directory (it does not support in-tree builds) and then building from
+that directory. To build systemd in a directory called `build` and run the tests:
+
+```sh
+$ meson setup build
+$ cd build
+$ meson compile
+$ meson test
```
Comment (daandemeyer) [inline docs/HACKING.md:21]: Meson can change directories for us. Also, the setup command is the default if none is specified so we can leave it out.

```sh
$ meson build
$ meson compile -C build
$ meson test -C build
```

Comment (daandemeyer) [reply docs/HACKING.md:21]: Actually, the `compile` command is rather new and not available on Ubuntu Focal yet. I'd just use ninja for compilaton/test here then:

```sh
$ meson build
$ ninja -C build
$ ninja -C build test
```

Comment (mbiebl) [reply docs/HACKING.md:21]: regarding `ninja test` vs `meson test`: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=912560

**Comment by poettering** [comment]
A few lines further down we give a much more complete run-through. Why isn't that enough?

**Comment by poettering** [comment]
Let's close this, the more complete list of commands further down I think covers this already nicely and is more comprehensive to get people stared.

**Pull request closed without merging**